students/rmart300/mailroom/mailroom.py

```diff
@@ -45,38 +56,45 @@ def create_report():
         donation_count = len(amount_list)
         donation_total = sum(int(v) for v in amount_list)
         donation_average = round(donation_total/donation_count,2)
-        data_row = "{0:20}  ${1:>15}   {2:>10}   ${3:>15}".format(donor, str(donation_total), str(donation_count), str(donation_average))
+        data_row = "{0:20}  ${1:>15}   {2:>10}   ${3:>15}".format(donor, 
+            str(donation_total), str(donation_count), str(donation_average))
         print(data_row)
 
 def write_letters():
-   for donor,name_dict in donor_dict.items():
-       first_name = name_dict['firstname']
-       last_name = name_dict['lastname']
-       amount_list = donation_dict[donor]
-       donation_num = 1
-       for donation_amt in amount_list:      
-           message = f"Dear {first_name} {last_name},\n\n    Thank you for your very kind donation of ${donation_amt}.\n\n    It will be put to very good use.\n\n    Sincerely,\n        -The Team"
-           with open(f"{letter_directory}{first_name}_{last_name}_{donation_num}.txt",'w') as f:
-               f.write(message)
-           donation_num += 1
+    for donor,name_dict in donor_dict.items():
+        first_name = name_dict['firstname']
+        last_name = name_dict['lastname']
+        amount_list = donation_dict[donor]
+        donation_num = 1
+        for donation_amt in amount_list:      
+            message = f"Dear {first_name} {last_name},\n\n    "
+            message += f"Thank you for your very kind donation of ${donation_amt}.\n\n    "
+            message += f"It will be put to very good use.\n\n    Sincerely,\n"
+            message += f"        -The Team"
+            with open(f"{letter_directory}{first_name}_{last_name}_{donation_num}.txt",'w') as f:
+                f.write(message)
+            donation_num += 1
 
 if __name__ == '__main__':
-    #build initial dictionary of donors and donation amounts
-    donor_list = ['Fred Smith','Terrie Ann','Murray Martin','Josh Jones','Jane Doe']
+    # build initial dictionary of donors and donation amounts

```

Comment:
I would recommend moving all your initial data structure setup logic to the top of the file where you define the data structures.

I think this would make your code more readable, and because the `if __name__ == '__main__':` block only executes when the file is run as a program, this would allow the module to be imported along with the initial data that you need to call your functions.